core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -328,7 +315,7 @@ private Object primaryReturnHandler(InvocationContext ctx, VisitableCommand visi
       }));
    }
 
-   private RpcOptions determineRpcOptionsForBackupReplication(RpcManager rpc, boolean isSync, List<Address> recipients) {
+   private RpcOptions determineRpcOptionsForBackupReplication(RpcManager rpc, boolean isSync, Collection<Address> recipients) {
```

Comment:
AFAICT `defaultSyncOptions` uses `ResponseMode.SYNCHRONOUS`, not `ResponseMode.SYNCHRONOUS_IGNORE_LEAVERS`.